languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -0,0 +1,244 @@
+;;;;;;;;;;;;;;;;;;;
```

Comment:
At the moment rules for different kinds of syntax nodes are mixed. Personally, I like to organize them in a top-down matter. So, that would be something like the following:

- program
- [top-level declarations](https://github.com/tree-sitter/tree-sitter-java/blob/master/grammar.js#L639): classes and class body, interfaces.
- [member declarations](https://github.com/tree-sitter/tree-sitter-java/blob/master/grammar.js#L833): field and method declarations
- [statements](https://github.com/tree-sitter/tree-sitter-java/blob/master/grammar.js#L427): blocks, local variables, expression statements, ...
- [expressions](https://github.com/tree-sitter/tree-sitter-java/blob/master/grammar.js#L170): variable references, method calls, ...
- comments at the end, as they are already

I have the feeling the grammar is a bit too liberal. A program consists of [statements](https://github.com/tree-sitter/tree-sitter-java/blob/master/grammar.js#L79), and [statements include all declarations](https://github.com/tree-sitter/tree-sitter-java/blob/master/grammar.js#L428). I think this is wrong, e.g. I don't think module, package, or import declarations are allowed in e.g. a method body, but only at the top-level. But perhaps this is not a big problem for the TSG.

Reply:
I added sections for each of the bullet points you mentioned (and one extra - "primary expressions") and reordered the rules according to the order they were declared in the grammar. Thank you for calling this out and linking to the grammar 😄 https://github.com/github/stack-graphs/pull/152/commits/15fefd30ed1d10bb6227bede4c75976a00151863